server/agent_streamable.go

```diff
@@ -401,6 +505,17 @@ func (a *OpenAICompatibleAgentImpl) executeToolCallsWithEvents(ctx context.Conte
 			continue
 		}
 
+		toolCtx := a.createToolContext(taskID, contextID)
+
+		var tool Tool
+		if a.toolBox != nil {
+			if tb, ok := a.toolBox.(*DefaultToolBox); ok {
+				if t, exists := tb.tools[toolCall.Function.Name]; exists {
+					tool = t
+				}
+			}
+		}
```

Comment:
We should have a `GetTool` method on the interface which would make this a lot nicer

Reply:
> We should have a `GetTool` method on the interface which would make this a lot nicer

Makes a lot of sense! will add it.